nengo/solvers.py

```diff
@@ -404,8 +408,10 @@ def __call__(self, A, Y, rng=None, E=None):
         import scipy.optimize
 
         tstart = time.time()
-        Y, m, n, d, matrix_in = format_system(A, Y)
-        Y = self.mul_encoders(Y, E)
+        Y, m, n, _, matrix_in = format_system(A, Y)
+        Y = self.mul_encoders(Y, E, copy=True)
+        d = Y.shape[1]
+        Y[Y < 0] = 0  # makes the final RMS error more reasonable
```

Comment:
I'm not sure we should be doing this.  The RMS error is what it is -- this is returning the RMS error where any negative target value is treated as if its target value was 0, so it's not really the RMS error any more.  Having this happen silently seems a bit odd to me...